packages/venia-ui/lib/components/App/container.js

```diff
@@ -1,39 +1,23 @@
-import React, { Component } from 'react';
-import { connect } from '@magento/venia-drivers';
+import React from 'react';
+import { useAppContext } from '@magento/peregrine/lib/context/app';
+import { useErrorContext } from '@magento/peregrine/lib/context/unhandledErrors';
 
-import appActions, { closeDrawer } from '../../actions/app';
 import App from './app';
-
-class AppContainer extends Component {
-    static get initialState() {
-        return {
-            renderError: null
-        };
-    }
-
-    static getDerivedStateFromError(renderError) {
-        return { renderError };
-    }
-
-    state = AppContainer.initialState;
-
-    render() {
-        const { renderError } = this.state;
-        return <ConnectedApp renderError={renderError} />;
-    }
-}
-
-const mapStateToProps = ({ app, unhandledErrors }) => ({
-    app,
-    unhandledErrors
-});
-
-const { markErrorHandled } = appActions;
-const mapDispatchToProps = { closeDrawer, markErrorHandled };
-
-const ConnectedApp = connect(
-    mapStateToProps,
-    mapDispatchToProps
-)(App);
+import { useErrorBoundary } from './useErrorBoundary';
+
+const AppContainer = () => {
```

Comment:
Just curious about why `ErrorBoundary` is special and gets its own file. We also may want to not name this file `container.js` because that's traditionally been for connecting components to Redux and may cause confusion.